### ISSUE TITLE
Can this be modified to support Azure OpenAI endpoints?

### ISSUE THREAD

**Issue by philipsd6** [charmbracelet/mods#73]
To use the Azure Cognitive services OpenAI endpoints, additional details are required. 

    OPENAI_API_TYPE="azure" # might need to support azure_ad, azuread as well, via OAuth...
    OPENAI_API_VERSION="2023-05-15" # OR 2021-11-01-preview, or...
    OPENAI_API_BASE="<The URL of your resource>" # See below...
    OPENAI_API_ENGINE="<The name of your deployment>"

For example, you might have an Azure OpenAI Resource named "My-Example-Cognitive-Resource" and in that resource you have deployed a gpt-35-turbo mode named "MyGPT35". [^1]

From that, the URI constructed for the `chat.completions` object will be:

https://my-example-cognitive-resource.openai.azure.com/openai/deployments/MyGPT35/chat/completions?api-version=2023-05-15

If I add what I can from the above to the mods config via `mods --settings` like this:

```yaml
apis:
  azure:
    base-url: https://my-example-cognitive-resource.openai.azure.com
    models:
      MyGPT35:
        aliases: ["35t"]
        max-input-chars: 12250
        fallback:
```

And then call:

    cat README.md | mods --api azure --model MyGPT35 "write a new section to this README documenting a Azure support feature"

It tries to process, but ends up returning:

    ERROR  Unknown OpenAI API error.
    error, status code: 404, message: Resource not found

Probably without support for Azure handling, `mods` is constructing the URL as:

https://my-example-cognitive-resource.openai.azure.com/engines/MyGPT35/chat/completions

instead of the correct sample above.

[^1]: This is a fake resource or deployment, just used as an example!

**Comment by philipsd6** [comment]
My bad, I see this is already being discussed and worked on in #58

**Comment by philipsd6** [comment]
Actually I'll reopen this to get some feedback, as I have concerns about the config file usage -- for when mods is finally supporting the use of `openai.DefaultAzureConfig` for Azure endpoints. 

If I have:
```yaml
apis:
  azure:
    base-url: https://my-example-cognitive-resource.openai.azure.com
    models:
      gpt-3.5-turbo:
        aliases: ["azgpt35t"]
        max-input-chars: 12250
        fallback:
  openai:
    base-url: https://api.openai.com/v1
    models:
      gpt-3.5-turbo:
        aliases: ["35t"]
        max-input-chars: 12250
        fallback:
```

My expectation was that I could use the `--api` parameter to point to the `azure` subtree, and then use `--model` to point to `gpt-3.5-turbo` and have it use the Azure base-url. But it does not. The documentation does not make clear what the proper use of `--api` is. 

However, since I can give them different aliases, I can at least point `--model` to `azgpt35t` and it will pull in the correct configuration. But although I've adjusted the`mods.go` file to call `ccfg = openai.DefaultAzureConfig(key, api.BaseURL)` when I use that alias (see #74) the error I get is missing the API:

    ERROR  Missing model 'azgpt35' for API ''
    error, status code: 404, message: The API deployment for this resource does not exist. If you created the deployment
    within the last 5 minutes, please wait a moment and try again.

I'm not a go programmer, so I have no idea if what I'm doing makes sense to anyone other than me.

**Comment by toby** [comment]
Thanks for all of this digging! I'm going to try to take a look at it this week and see if we can get something that works well with both apis and isn't too confusing.

**Comment by sheldonhull** [comment]
Been trying to figure this out as well. Is anyone actually using azure openai successfully with the tool after the latest PR? I'm getting Resource not found as well.

**Comment by toby** [comment]
@sheldonhull we haven't released it yet so if you're installing with a package manager you won't haven openai support. If you build it from `main` it should work though. This is what the config should [look like](https://github.com/charmbracelet/mods/pull/58#issuecomment-1599696274). You can run `mods --reset-settings` to generate that.

**Comment by philipsd6** [comment]
Yes I've built it from main and it works fine with our Azure cognitive services deployment with the proper configuration.

**Comment by toby** [comment]
Thanks for the confirmation @philipsd6. @sheldonhull my plan was to close this issue when we do our next release (coming soon). Let me know if you're still having an error before that and we can try to figure it out.

**Comment by sheldonhull** [comment]
Great help. I was using aqua tool to go install latest but I think from github that was 0.2.0 and not main ad I ping all binary tools. 

Very thankful for this and look forward to trying out a lot more options now that I can use internally. Cheers.

**Comment by toby** [comment]
We've officially shipped Azure OpenAI support in [v1.0.0](https://github.com/charmbracelet/mods/releases/tag/v1.0.0).
